pkg/tfbridge/provider.go

```diff
@@ -59,6 +59,63 @@ type Resource struct {
 	TFName string           // the Terraform resource name.
 }
 
+// collectImporterAttributes runs the Terraform Importer defined on the Resource for the given
+// resource ID, and adds any non-ID attributes to the inputs map.
+func (res *Resource) collectImporterAttributes(resourceID resource.ID, inputs map[string]string) error {
+	// There is nothing to do here if the resource doesn't have an importer defined in the
+	// Terraform schema.
+	if res.TF.Importer == nil {
+		return nil
+	}
+
+	id := resourceID.String()
+
+	glog.V(9).Infof("%s has TF Importer", res.TFName)
+
+	// Prepare a Terraform ResourceData for the importer
+	data := res.TF.Data(nil)
+	data.SetId(id)
+	data.SetType(res.TFName)
+
+	results, err := res.TF.Importer.State(data, nil)
+	if err != nil {
+		return errors.Wrapf(err, "importing %s", id)
+	}
+
+	if len(results) < 1 {
+		return errors.Errorf("importer for %s returned no resources", id)
```

Comment:
Any idea when this is possible with `err == nil`?

Reply:
I don't _think_ it is possible, but I'm not sure relying on every importer behaving correctly here when the alternative is a panic is such a good idea?

Reply:
>  the alternative is a panic is such a good idea

The alternative I was considering was just returning `nil` and letting `Refresh` do its thing. Not sure if that's better, though.

Reply:
That could be a valid alternative - though if the resource is buggy enough that a defined importer doesn't return a state, I'm not sure what we can infer from that.

I'm happy to defer to your knowledge of the codebase here - which do you think is a better approach here given how we handle similar situations elsewhere?

Reply:
Offline discussion determined we should error here requesting a bug report in the same manner as Terraform does in the face of a nil state.